File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        def isAri(nums):
            if len(nums) <= 1:
                return True
            nums.sort()
            diff = nums[1] - nums[0]
            for i in range(2,len(nums)):
                if nums[i] - nums[i-1] != diff:
                    return False
            return True
        
        ans = []
        for i in range(len(l)):
            ans.append(isAri(nums[l[i]:r[i]+1]))
        return ans

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> ans = new ArrayList<>();
        for(int i = 0; i < l.length; i++){
            ans.add(isAri(nums, l[i] , r[i] ));
        }
        return ans;
    }
    private boolean isAri(int[] nums, int start, int end){
        if(end - start + 1 <= 1) return true;
        int[] temp = new int[end - start + 1];
        for(int i = start; i <= end; i++){
            temp[i-start] = nums[i];
        }
        Arrays.sort(temp);
        int diff = temp[1] - temp[0];
        for(int i = 2; i < temp.length; i++){
            if(temp[i] - temp[i-1] != diff) return false;
        }
        return true;
    }

}